### ISSUE TITLE
Deleting saved searches from the TC does not work

### ISSUE THREAD

**Issue by camerojo** [talentbeyondboundaries/tbbtalentv2#1606]
The user sees a "You cannot delete this saved search/list message". 
Reported by Mari. See https://refugeejobsmarket.slack.com/archives/C42S4AJ73/p1730468635110609

The issue is that a saved Search will be referred in table saved_list to by one (or more if the search is shared with others) selection lists. If it is starred by other users it can also appear in user_saved_search.

**Comment by samschlicht** [comment]
- Another user or I can star or watch it and it can still be deleted
- I can make it reviewable and thereby referenced in saved list and it can still be deleted
- I can make a suggested search for a job and still delete it
 
I'll need to check with John what I'm missing here.

**Comment by samschlicht** [comment]
Just noting that I spoke to John and he can't replicate this either! But there's obviously _somewhere_ that a saved search can be referenced that isn't handled in the delete code. I'll do another scan when I get back to this one.

**Comment by camerojo** [comment]
@samschlicht and I figured this one out. I sent us off on a wild goose chase with my initial comments in the description because in fact the code does NOT actually delete searches. It just marks their status as deleted.

The problem was because we changed the minimum DTO for uploaded searches to not include createdBy - which meant that the Angular code could not identify who owned the search.

createdBy has been added in as the minimum DTO for searches and lists in an earlier bug fix in staging (which is why we couldn't duplicate the error there). So this problem will be fixed in the next release to production.
